ipatests/test_integration/test_automember.py

```diff
@@ -0,0 +1,366 @@
+            hostgroup2 = 'Boston'
+            host1 = 'host1.example.com'
+
+            self.automember_hostgroup_add(hostgroup1)
+            self.automember_hostgroup_add(hostgroup2)
+            self.host_automember_setup(hostgroup1, host1)
+
+            result = self.master.run_command(
+                ['ipa', 'hostgroup-show', hostgroup1]
+            )
+            assert host1 in result.stdout_text
+
+            # Modifying the user group so that it becomes part of new group
+            self.host_automember_mod(hostgroup2, host1)
+
+            result = self.master.run_command(
+                ['ipa', 'hostgroup-show', hostgroup2]
+            )
+            assert host1 in result.stdout_text
+
+            result = self.master.run_command(
+                ['ipa', 'hostgroup-show', hostgroup1]
+            )
+            assert host1 not in result.stdout_text
+
+            # Deleting the user entry and checking its not part of group now
+            self.host_automember_cleanup(host1)
+
+            result = self.master.run_command(
+                ['ipa', 'hostgroup-show', hostgroup2]
+            )
+            assert host1 not in result.stdout_text
+
+        finally:
+            # testcase cleanup
+            self.automember_hostgroup_cleanup(hostgroup1)
+            self.automember_hostgroup_cleanup(hostgroup2)
+
+    def test_modify_user_entry_unmembership_disabled(self):
+        # https://pagure.io/389-ds-base/issue/50077
+        """
+        Test that an update on a user entry does not triggers re-evaluation of
+        the user's automember groups when auto-unmembership is disabled. We
+        need to run automembership-rebuild for correct membership of user.
+        """
+        try:
+            master = self.master
+            tf = NamedTemporaryFile()
+            ldif_file = tf.name
+
+            entry_ldif = textwrap.dedent("""
+                dn: cn=Auto Membership Plugin,cn=plugins,cn=config
+                changetype: modify
+                replace: automemberprocessmodifyops
+                automemberprocessmodifyops: off""")
+
+            master.put_file_contents(ldif_file, entry_ldif)
+
+            arg = ['ldapmodify',
+                   '-h', master.hostname,
+                   '-p', '389', '-D',
+                   str(master.config.dirman_dn),   # pylint: disable=no-member
+                   '-w', master.config.dirman_password,
+                   '-f', ldif_file]
+
+            master.run_command(arg)
+            master.run_command(['ipactl', 'restart'])
+            search_cmd = [
+                'ldapsearch', '-x',
+                '-D', 'cn=directory manager',
+                '-w', master.config.dirman_password,
+                '-b', 'cn=Auto Membership Plugin,cn=plugins,cn=config',
+                '-s', 'base']
+
+            result = self.master.run_command(
+                search_cmd,
+                raiseonerr=False
+            )
+            assert result.returncode == 0
```

Comment:
There is no need to check returncode, it is better call run_command without "raiseonerr=False" -- in case of error it will print useful debug info like invoked command, actual return code and stdout/stderr.